### PR TITLE
Modify user attributes (home, gecos, shell and groups)

### PR DESCRIPTION
**Pull request by imobachgs**
Bring back be876b7d0cbc28382f6f92c818a645788a1ac043 which added support for `usermod` and was wrongly removed by me when adapting the original PR.

### CODE REVIEW

File: src/lib/y2users/linux/writer.rb [hunk 1]
```diff
@@ -271,6 +273,30 @@ def set_password_attributes(user, issues)
...
+          !new_user.public_send(attr).nil? &&
+            (new_user.public_send(attr) != old_user.public_send(attr))
+        end
+
+        Yast::Execute.on_target!(USERMOD, *usermod_options(new_user, old_user)) if usermod_changes
+        change_password(new_user, issues) if old_user.password != new_user.password
+        write_auth_keys(new_user, issues) if old_user.authorized_keys != new_user.authorized_keys
+      rescue Cheetah::ExecutionFailed => e
+        issues << Y2Issues::Issue.new(
+          format(_("The user '%{username}' could not be modified"), username: new_user.name)
```
Comment (jreidinger) [inline src/lib/y2users/linux/writer.rb:295]: Not related to this PR, but it is quite useless message without details. What about support for details in issues and add to that stderr where is usually useful details why it failed?

**Comment by jreidinger** [comment]
sadly what is also missing and was removed originally is support for groups - https://github.com/yast/yast-users/pull/272/commits/913cb87223185c8a1ef4c4562d8299c971cbc15c and https://github.com/yast/yast-users/pull/272/commits/904a6c7a5a55715077815cf51355497d26ecfd99 it is needed as autoyast support groups and additional groups

**Comment by imobachgs** [comment]
> sadly what is also missing and was removed originally is support for groups - [913cb87](https://github.com/yast/yast-users/commit/913cb87223185c8a1ef4c4562d8299c971cbc15c) and [904a6c7](https://github.com/yast/yast-users/commit/904a6c7a5a55715077815cf51355497d26ecfd99) it is needed as autoyast support groups and additional groups

Well, it is not exactly missing. As it is only supported by AutoYaST, we decided to move that logic to the `autoinst` reader. Actually, even the `groups` key was removed from `useradd`. I have a task in my todo list related to that.

**Comment by imobachgs** [comment]
> > sadly what is also missing and was removed originally is support for groups - [913cb87](https://github.com/yast/yast-users/commit/913cb87223185c8a1ef4c4562d8299c971cbc15c) and [904a6c7](https://github.com/yast/yast-users/commit/904a6c7a5a55715077815cf51355497d26ecfd99) it is needed as autoyast support groups and additional groups
> 
> Well, it is not exactly missing. As it is only supported by AutoYaST, we decided to move that logic to the `autoinst` reader. Actually, even the `groups` key was removed from `useradd`. I have a task in my todo list related to that.

mmm perhaps I missread your comment. Anyway, I will introduce it in a different PR.

File: src/lib/y2users/linux/writer.rb (hunk 1 above)
Comment (imobachgs) [reply src/lib/y2users/linux/writer.rb:295]: Yes, it is true. However, issues do not contain "details", so it is perhaps something to improve. Actually, you can find the details in the logs.

File: test/lib/y2users/config_merger_test.rb [hunk 2]
```diff
@@ -194,10 +194,10 @@ def lhs_group(name)
           expect(lhs.groups.map(&:name)).to contain_exactly("test1", "test2", "test3")
         end
 
-        it "updates the lhs group with the data from the corresponding rhs" do
+        it "updates the lhs group with the data from the corresponding rhs group except gid" do
           subject.merge
 
-          expect(lhs_group("test1")).to eq(group1)
+          expect(lhs_group("test1").gid).to eq(100)
```
Comment (jreidinger) [inline test/lib/y2users/config_merger_test.rb:200]: this is probably left over from my commit as it should not change merger behavior, so test should fail.

**Review by jreidinger — APPROVED** [review]

**Pull request merged**